application/src/main/java/org/togetherjava/tjbot/commands/help/AskCommand.java

```diff
@@ -102,8 +122,44 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         overviewChannel.createThreadChannel(name.toChannelName())
             .flatMap(threadChannel -> handleEvent(eventHook, threadChannel, author, title, category,
                     guild))
-            .queue(any -> {
-            }, e -> handleFailure(e, eventHook));
+            .queue(any -> userToLastAsk.put(event.getUser().getIdLong(), Instant.now()),
+                    e -> handleFailure(e, eventHook));
+    }
+
+    private boolean isUserOnCooldown(User user) {
+        return Optional.ofNullable(userToLastAsk.getIfPresent(user.getIdLong()))
+            .map(lastAction -> lastAction.plus(COOLDOWN_DURATION_VALUE, COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private void sendCooldownResponse(SlashCommandInteractionEvent event) {
+        User user = event.getUser();
+        Guild guild = event.getGuild();
+
+        RestAction<String> changeTitle = mentionHelpChangeCommand(guild, CHANGE_TITLE_SUBCOMMAND);
+        RestAction<String> changeCategory =
+                mentionHelpChangeCommand(guild, CHANGE_CATEGORY_SUBCOMMAND);
+        long lastThreadByAuthorId = database
+            .read(context -> context.selectFrom(HELP_THREADS)
+                .where(HELP_THREADS.AUTHOR_ID.eq(user.getIdLong()))
+                .orderBy(HELP_THREADS.CREATED_AT.desc())
+                .limit(1)
+                .fetchOne())
```

Comment:
Doesn't this just do the same?

Reply:
no `.fetchOne()` needs to limit the query to 1 result, if i have multiple results it'll throw an exception

Reply:
I see, my bad